packages/astro/src/executors/dev/executor.ts

```diff
@@ -99,7 +99,11 @@ function getAstroDevArgs(
     args.push('--drafts');
   }
   if (options.host !== undefined) {
-    args.push('--host', options.host.toString());
+    if (typeof options.host === 'boolean' && options.host === true) {
+      args.push('--host');
+    } else {
+      args.push('--host', options.host.toString());
+    }
```

Comment:
We also need to handle `false` correctly:
```suggestion
    if (typeof options.host === 'boolean') {
      args.push(options.host === true ? '--host' : '--no-host');
    } else {
      args.push('--host', options.host.toString());
    }
```